src/hooks/use-click-outside/index.ts

```diff
@@ -0,0 +1,20 @@
+import { MutableRefObject, useEffect } from 'react'
+
+export const useClickOutside = (ref: MutableRefObject<HTMLDivElement | null>, onClickOutside: VoidFunction) => {
```

Comment:
**note:** Not something you need to change in this PR, but I think this hook might be better named `useClickOutsideListener` 🤔 Currently, `useClickOutside` could be taken to mean it allows a consumer to click outside an element or something... Not a big deal though and I think we're better off moving forward and leaving that rename for another time